platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -160,6 +166,18 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
         }
     }
 
+    private
```

Comment:
Did this move because it changed scope? It seems like something that CC should just use, rather than know how to create.

Reply:
It moved specifically here, because it actually should be a `BuildTree` scope, and it would be required by Kotlin DSL builders, and it wasn't visible for them when declared in the `ToolingModelServices`.

Is there a better place to declare a BuildTree service that comes before CC infra does service creation?
My understanding was that here we are not actually in CC land yet, because this class handles both vintage and CC cases. In other words, `DefaultBuildTreeModelControllerServices` is used unconditionally of whether CC is enabled, because it actually determines, whether CC is enabled.

Reply:
I think we should extract a new project in the core platform that contains the tooling services that run in the daemon. We can do that later, though.